egs/iam/s5/local/prepare_dict.sh

```diff
@@ -1,5 +1,16 @@
 #!/usr/bin/env bash
 
+""" This module prepares dictionary directory. It creates lexicon.txt,
```

Comment:
Ashish, I thought we stopped including the test words into the lexicon. Below I noticed `local/prepare_lexicon.py $train_text $test_text $dir` which means you are including the test words in the lexicon. I'm not  sure if it will have an effect in decoding (since you are not including the test sentences in the lm training code, right?) but to make it clear, let's remove this and test it. Also, try to do what prepare_lexicon.py does *inline* in this script (and remove prepare_lexicon.py).

Reply:
To do function of prepare_lexicon.py inline in the script. 
In the lexicon.txt file, in the second column, I am trying to replace '#' with '<HASH>' but not able to do so, it is replacing both '#', can you please help me with below script.

# reads the data from text files and write all unique words in lexicon.txt file
# find all unique words
cat $train_text/text | awk '{ for(i=2;i<=NF;i++) print $i;}' | sort -u >train_words.txt
cat $test_text/text | awk '{ for(i=2;i<=NF;i++) print $i;}' | sort -u >test_words.txt
filter_scp.pl --exclude train_words.txt test_words.txt >diff.txt
cat train_words.txt > all_words
cat diff.txt  >> all_words

# write words in following format: Ben B e n
awk '{
  printf("%s", $1);
    for(j=1;j<=length($1);++j) {
      printf(" %s", substr($1, j, 1));
    }
  printf("\n");                   
}' "all_words" | sort -k1 > lexicon_with_hash

#replace '#' with '<HASH>'
sed -i 's/\#/<HASH>/g' lexicon_with_hash > $dir/lexicon.txt

Reply:
Sorry, Test words were used in unk modeling i will remove the test words.